### DISCUSSION TITLE
Sharing of authentication in pipeline tasks

### DISCUSSION THREAD

**Discussion by imranakram** [microsoft/powerplatform-build-tools#495] [Ideas]
We have a lot of tasks in a pipeline that work on the same environment. The thing that becomes immediately evident is that it tries to authenticate in every task individually which makes me wonder if it would be possible to authenticate once and use across the whole pipeline as long as it's the same connection selected. I'm using the service principle option. That could potentially boost performance in such pipelines.

**Comment by petrochuk** [comment]
We use MSAL library for authentication. It has internal per user token cache, so authentication happens only once (per service) and cached token reused usually for an hour.

**Reply by imranakram** [reply]
Ok, that's great! Thanks for the detail :)
